app/controllers/peer_reviews_controller.rb

```diff
@@ -1,67 +1,167 @@
+    number_of_reviews_for_reviewer = {}
 
-    respond_to do |format|
-      if @peer_review.save
-        format.html { redirect_to @peer_review, notice: 'Peer review was successfully created.' }
-        format.json { render :show, status: :created, location: @peer_review }
-      else
-        format.html { render :new }
-        format.json { render json: @peer_review.errors, status: :unprocessable_entity }
+    reviewee_to_reviewers_map.each do |reviewee_id_key, list_reviewers_id_array|
+      list_reviewers_id_array.each do |reviewer_id|
+        unless number_of_reviews_for_reviewer.key?(reviewer_id)
+          number_of_reviews_for_reviewer[reviewer_id] = 0
+        end
+        number_of_reviews_for_reviewer[reviewer_id] += 1
       end
     end
+
+    return number_of_reviews_for_reviewer
   end
 
-  # PATCH/PUT /peer_reviews/1
-  # PATCH/PUT /peer_reviews/1.json
-  def update
-    respond_to do |format|
-      if @peer_review.update(peer_review_params)
-        format.html { redirect_to @peer_review, notice: 'Peer review was successfully updated.' }
-        format.json { render :show, status: :ok, location: @peer_review }
+  def assign_groups
+    @assignment = Assignment.find(params[:assignment_id])
+    selected_reviewer_group_ids = params[:selectedReviewerGroupIds]
+    selected_reviewee_group_ids = params[:selectedRevieweeGroupIds]
+    reviewers_to_remove_from_reviewees_map = params[:selectedReviewerInRevieweeGroups]
+    action_string = params[:actionString]
+
+    case action_string
+      when 'random_assign'
+        randomly_assign
+      when 'assign'
+        if selected_reviewer_group_ids.size == 0
+          render text: 'Cannot have an empty list of reviewers', status: 400
+          return
+        elsif selected_reviewee_group_ids.size == 0
+          render text: 'Cannot have an empty list of reviewees', status: 400
+          return
+        end
+        reviewer_groups = Grouping.where(id: selected_reviewer_group_ids)
+        reviewee_groups = Grouping.where(id: selected_reviewee_group_ids)
+        assign(reviewer_groups, reviewee_groups)
+      when 'unassign'
+        unassign(reviewers_to_remove_from_reviewees_map)
       else
-        format.html { render :edit }
-        format.json { render json: @peer_review.errors, status: :unprocessable_entity }
+        render text: 'Unexpected action type', status: 400
+        return
+    end
+
+    head :ok
+  end
+
+  def randomly_assign
+    # TODO
+  end
+
+  def assign(reviewer_groups, reviewee_groups)
+    # TODO - Do not allow assigning if the user is already assigned (maybe put unique constraints on the table instead?)
+    reviewer_groups.each do |reviewer_group|
+      reviewee_groups.each do |reviewee_group|
+        # TODO - is this okay to do? should the result be cached? or does rails do this for us transparently?
+        result = reviewee_group.current_submission_used.get_latest_result
```

Comment:
Well, this is okay for now, but we really should be creating a new result for each peer review.